### ISSUE TITLE
Question about the reasoning part in paper 

### ISSUE THREAD

**Issue by dydxdt** [UniModal4Reasoning/SimChart9K#3]
<img width="944" alt="image" src="https://github.com/UniModal4Reasoning/SimChart9K/assets/46140371/b9a94acf-804e-4bd3-95c5-115ca693c889">
Great job! 
I have a question about this part. What's the model used for reasoning? I read the paper and am not very sure about this. I suppose CIE is used for chart perception, so what's the model used in reasoning? It said GPT3.5, but I don't knwo how to use it to train for the downstream task? Can you explain it? Thx!

**Comment by renqiux0302** [comment]
Thanks for your recognition! 
We have employed `gpt-3.5-turbo` as the reasoning model, which is zero-shot without training on customized data. Also, we have recently updated our reasoning model with a trainable open-source language model, which we believe will be open-source in the near future.
Hoping you stay tuned to our work. And feel free to contact us if there is any concerns for you.

**Comment by dydxdt** [comment]
Thanks! got it!

**Comment by dydxdt** [comment]
<img width="1029" alt="image" src="https://github.com/UniModal4Reasoning/SimChart9K/assets/46140371/901cb443-6a6b-4db6-a91f-3cd304c59901">
I also have another question about the "self-inspection mechanism". Since I also have tried to use GPT4 to generate the drawing code and found it didn't always follow the instructions strictly and generated codes which couldn't run successfully, I am not clear about the mentioned "self-inspection mechanism". Can you explain more? Or can you give an example of the instructions? Thanks!

**Comment by renqiux0302** [comment]
Of course. The "self-inspection mechanism" here refers to that, in the initial drawing code generation loop, we will self-inspect the executability of the drawing code, and save the CSV texts corresponding to the unexecutable code. For the next code simulation generation loop, we only deal with the CSV texts saved in the previous loop. And so on until all the texts have executable drawing code pair. In our experiment, the code simulation process for `SimChart9k` can be completed in three loops.
As you mentioned, the prompts and instructions are quite difficult to design for chart simulation. You can refer to our templates in Figs 4 and 5 of Appendix B for your own design.

**Comment by dydxdt** [comment]
Thanks！
